### PR TITLE
Move easy THStorage functions out of generic.

### PR DESCRIPTION
**Pull request by ezyang**
I didn't move any functions which actually needed to generate
real-specific code.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

### CODE REVIEW

File: aten/src/TH/THStorage.cpp [hunk 1]
```diff
@@ -96,3 +96,145 @@ THStorage* THStorage_newWithAllocator(at::ScalarType scalar_type, ptrdiff_t size
...
+
+void THStorage_clearFlag(THStorage *storage, const char flag)
+{
+  storage->flag &= ~flag;
+}
+
+void THStorage_retain(THStorage *storage)
+{
+  if(storage && (storage->flag & TH_STORAGE_REFCOUNTED))
+    ++storage->refcount;
```
Comment (gchanan) [inline aten/src/TH/THStorage.cpp]: nit: when I've done these moves, I've added braces in this situation to match the way most of our ATen/pytorch C++ code is written.

**Review by gchanan — APPROVED** [review]
lgtm!

File: aten/src/TH/THStorage.cpp (hunk 1 above)
Comment (ezyang) [reply aten/src/TH/THStorage.cpp]: easy enough to fix

**Comment by ezyang** [comment]
Rolled into #9136

**Pull request closed without merging**